Constructive Code Review/outline.txt

```diff
@@ -0,0 +1,205 @@
+              excited about the actual feature it delivers, before I
+              dig in and discover the code is terrible. "Aw, man, I
+              can't wait to land this. I'm going to love being able to
+              close Flash tabs without reaching for the mouse. [Now
+              let's get the problems fixed so we can merge it.]"
+            - "Good catch!"
+                  These comments would make a good visual, as actual
+                  line comments on a PR.
+                - Also makes a good reply back to the reviewer
+            - "Thank you so much for refactoring this scary mess."
+        - Emoticons
+        - GIFs
+    - TLDR;LGTM
+          A 1000-line review gets 2 comments. A 10-line one gets 5.
+        - Feel free to ask for a more reviewable commit structure,
+          better commit messages, more comments, a couple of paragraphs
+          of prose explaining the approach. If you can't understand it
+          now (with somebody around to answer questions), just imagine
+          the trouble in a year.
+            - When you submit a patch, structure it so you'd be happy
+              to review it.
+            - And definitely read your diffs first before submitting.
+                - Hugely worth it to get a diff viewer you like
+        - Consider not burning yourself out on it; some studies have
+          found a decrease in review effectiveness after about an hour.
+    - Fix the rest of this file while you're at it.
+        - Don't expect them to fix the whole module if they touch part
+          of it.
+            - The idea is for the codebase to be GettingBetter, not
+              BeingPerfect. You can always come back and fix other
+              parts later, and it's okay to point that out, but make
+              clear that it doesn't block the merge (and shouldn't
+              distract them).
+    - Nitpicks
+        - Having a style guide or a linter smooths things along, saves
+          back-and-forth.
+            - But overzealous linters just make me angry. Code is for
+              humans.
+        - We have a bunch of decent style guides at Mozilla; you should
+          just adopt them when possible and save everybody
+          work/inconsistency.
+        - The webdev ones are just in a GH repo. If you don't like 'em,
+          submit a PR.
+        - Style: consistency within a function, then within a file,
+          then within a module, then within a project. Try not to be
+          adaptable. You can always do your favorite style on your
+          personal stuff.
+        - Remember it costs something like $100/hr to have you sit
+          there, and throttle accordingly.
+    - Sooner is better
+        - Even if you aren't comprehensive
+            - You can say “I'm not going to have the time to look at
+              the rest of this for a week. Someone else can feel free
+              to step in.” Don't feel like you married the patch.
```

Comment:
At a bare minimum, it's important to reply fast and set expectations. I think that's sort of covered in this one, but with different phrasing maybe?

Plus if you're in a position where you're the only reviewer, delegate as much as humanly possible wherever you can delegate it.